main/main.zkasm

```diff
@@ -2,39 +2,40 @@ INCLUDE "constants.zkasm"
 INCLUDE "vars.zkasm"
 
 ; Blocks zkROM
-;       A - Load input variables
+;       A - Load initial registers into memory: oldStateRoot (B), oldAccInputHash (C), oldNumBatch (SP) & chainID (GAS)
 ;       B - Set batch global variables
 ;       C - Loop parsing RLP transactions
 ;       D - Loop processing transactions
-;       E - Batch asserts: localExitRoot, transactions size, batchHashData & globalHash
+;       E - Batch computations: get newLocalExitRoot, assert transactions size, compute batchHashData & compute newAccInputHash
 ;       F - Finalize execution
 
 start: ; main zkROM entry point
 ;;;;;;;;;;;;;;;;;;
 ;; A - Load input variabales
 ;;;;;;;;;;;;;;;;;;
-        ${getGlobalHash()}                  :MSTORE(globalHash)
         STEP => A
-        1                                   :ASSERT ; Ensure it is the begining of the execution
+        0                                   :ASSERT ; Ensure it is the beginning of the execution
+        B                                   :MSTORE(oldStateRoot)
+        C                                   :MSTORE(oldAccInputHash)
+        SP                                  :MSTORE(oldNumBatch)
+        GAS                                 :MSTORE(chainID)
+
         ${getGlobalExitRoot()}              :MSTORE(globalExitRoot)
-        ${getOldStateRoot()}                :MSTORE(oldStateRoot)
-        ${getOldLocalExitRoot()}            :MSTORE(oldLocalExitRoot)
         ${getSequencerAddr()}               :MSTORE(sequencerAddr)
-        ${getBatchHashData()}               :MSTORE(batchHashData)
-        ${getNumBatch()}                    :MSTORE(numBatch)
         ${getTimestamp()}                   :MSTORE(timestamp)
-        ${getChainId()}                     :MSTORE(chainId)
         ${getTxsLen()}                      :MSTORE(batchL2DataLength)
 
-        ; Fill globalHash: oldStateRoot & oldLocalExitRoot
+        B => SR ;set initial state root
+
+        ; Fill newAccHashInput: oldAccInputHash
         32 => D
-        $ => A                              :MLOAD(oldStateRoot)
-        A                                   :HASHK(0) ; GlobalHash address is 0
-        A => SR
-        $ => A                              :MLOAD(oldLocalExitRoot)
-        A                                   :HASHK(0)
+        $ => A                              :MLOAD(oldAccInputHash)
+        A                                   :HASHK(0) ; 'newAccHashInput' hashK address is 0
```

Comment:
This parameter is in C, no need to load it

Reply:
I think it is better to compute all the `newAccHashInput` at the end for clarity. So this comment does not longer apply.